Packs/PerimeterX/Integrations/BotDefender/BotDefender.py

```diff
@@ -0,0 +1,361 @@
+    """
+
+    # Check for the field to query for
+    search_type = args.get('search_type', None)
+    if not search_type:
+        raise ValueError('No search_type specified')
+
+    # Check to make sure we have a query term
+    search_term = args.get('search_term', None)
+    if not search_term:
+        raise ValueError('No search_term specified')
+
+    # Check for an IP based investigation
+    supported_ip_search_types = ['true_ip', 'socket_ip']
+
+    # risk_score=result['maxRiskScore']
+
+    if search_type in supported_ip_search_types:
+        """
+        Run an IP based search if the search type is one supported by the IP types
+        """
+        result = client.post_investigate_by_ip(ip_type=search_type, ip_address=search_term)
+
+        pXScore = pXScoring()
+        
+        pXScore.add_score_to_ip_list(ip_address=search_term, 0, thresholds=thresholds)
+
+        return CommandResults(
+            outputs_prefix='PerimeterX',
+            outputs_key_field='',
+            outputs=result,
+            indicators=pXScore.ip_standard_list
+        )
+
+    elif search_type == 'name':
+        """
+        THIS IS NOT CURRENTLY IMPLEMENTED
+        Run a name based search if the search type is name
+        """
+        result = client.post_investigate_by_name(name_type=search_type, name=search_term)
+
+    else:
+        """
+        Generate an error because the search type is not supported
+        """
+        raise ValueError('Invalid search_type provided')
+
+    readable_output = f'{result}'
+
+    return CommandResults(
+        outputs_prefix='PerimeterX',
+        outputs_key_field='',
+        outputs=result
+    )
+
+
+def ip(client: Client, args: Dict[str, Any], thresholds: Dict[str, Any]):
+    """
+    Collect the details to run an IP Reputation query against the PerimeterX API
+
+    :type ip: ``str``
+    :param ip: Results will be provided for this particular IP
+
+    :return:
+        A ``CommandResults`` object that is then passed to ``return_results``,
+        that contains Domains
+
+    :rtype: ``CommandResults``
+    """
+
+    # Check to make sure we have a query term
+    ip_address = args.get('ip', None)
+    if not ip_address:
+        raise ValueError('No IP Address specified')
+
+    result = client.post_investigate_by_ip(ip_type='true_ip', ip_address=ip_address)
+
+    pXScore = pXScoring()
+
+    pXScore.add_score_to_ip_list(ip_address=ip_address, risk_score=result['maxRiskScore'], thresholds=thresholds)
+
+    readable_output = f'{pXScore.ip_standard_list}'
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='PerimeterX',
+        outputs_key_field='',
+        outputs=result,
+        indicators=pXScore.ip_standard_list
+    )
+
+
+def main():
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+    api_key = demisto.params().get('apikey')
+
+    # get the service API url
+    # '/investigate?search=ip:1.1.1.1&tops=path' + api_key
```

Comment:
What about this?

Reply:
This should be received by contacting PX